src/components/UserDashboard/AuthorizedStudies/Gen3Connected.js

```diff
@@ -98,23 +113,29 @@ const Gen3Connected = ({
       ) : (
         <Column>
           {authorizedStudies ? (
-            authorizedStudies.map(({ studyShortName, id: studyId }) => {
-              const { authorizedFiles, unAuthorizedFiles, consentCodes } = combinedStudyData[
-                studyId
-              ];
-              return (
-                <Study
-                  key={studyId}
-                  studyId={studyId}
-                  name={studyShortName}
-                  consentCodes={consentCodes}
-                  authorized={authorizedFiles.length}
-                  total={authorizedFiles.length + unAuthorizedFiles.length}
-                  onStudyTotalClick={onStudyTotalClick(studyId)}
-                  onStudyAuthorizedClick={onStudyAuthorizedClick(studyId)}
-                />
-              );
-            })
+            !authorizedStudies.length ? (
+              <Box mt={20}>
+                <NoAuthorizedStudiesMessage />
+              </Box>
+            ) : (
+              authorizedStudies.map(({ studyShortName, id: studyId }) => {
+                const { authorizedFiles, unAuthorizedFiles, consentCodes } = combinedStudyData[
```

Comment:
if no `combinedStudyData[studyId]` then all these vars will be undefined, can't call `.length` on undefined, need check?

Reply:
Variable is built in this file so shouldn't be a problem.